src/bokeh/core/property/container.py

```diff
@@ -202,11 +202,14 @@ def validate(self, value: Any, detail: bool = True) -> None:
 
         key_is_valid = self.keys_type.is_valid
         value_is_valid = self.values_type.is_valid
-        if isinstance(value, dict) and all(key_is_valid(key) and value_is_valid(val) for key, val in value.items()):
-            return
-
-        msg = "" if not detail else f"expected an element of {self}, got {value!r}"
-        raise ValueError(msg)
+        expected = f"expected a dict of type {self}"
+        if not isinstance(value, dict):
+            raise ValueError(f"{expected}, got a value of type {type(value)}" if detail else "")
+        if any(bad_keys := [str(k) for k in value if not key_is_valid(k)]):
+            raise ValueError(f"{expected}, got a dict with invalid keys: {','.join(bad_keys)}" if detail else "")
```

Comment:
```suggestion
        expected = f"expected a dict of type {self}"
        
        if not isinstance(value, dict):
            raise ValueError(f"{expected}, got a value of type {type(value)}" if detail else "")
            
        if any(bad_keys := [str(k) for k in value if not key_is_valid(k)]):
            raise ValueError(f"{expected}, got a dict with invalid keys: {','.join(bad_keys)}" if detail else "")
```
@marceloslacerda this is what I meant by vertical space: visual separation of blocks of code